fnf/data/query_mag_composite.py

```diff
@@ -0,0 +1,51 @@
+import requests
+from retrying import retry
+
+ENDPOINT = "https://api.labs.cognitive.microsoft.com/academic/v1.0/evaluate"
+
+
+def build_composite_expr(query_values, entity_name, year):
+    """Builds a composite expression with ANDs in OR to be used as MAG query.
+
+    Args:
+        query_values (:obj:`list` of str): Phrases to query MAG with. 
+        entity_name (str): MAG attribute that will be used in query.
+        year (int): We collect data in the [year, now] timeframe.
+    Returns:
+        (str) MAG expression.
+    
+    """
+    query_prefix_format = "expr=OR({})"
+    and_queries = [
+        "".join([f"And(Composite({entity_name}='{query_value}'), Y>={year})"])
+        for query_value in query_values
+    ]
+    return query_prefix_format.format(", ".join(and_queries))
+
+
+@retry(stop_max_attempt_number=1)
```

Comment:
Does this try once or retry once?
If it only tries once then can we get rid of this?

Reply:
Missed that. I lowered it to 1 so that I can fail fast during testing the API.